runtime/hydradx/src/assets.rs

```diff
@@ -1845,7 +1845,12 @@ impl SwappablePaymentAssetTrader<AccountId, AssetId, Balance> for XykPaymentAsse
 		asset_out: AssetId,
 		asset_out_amount: Balance,
 	) -> Result<Balance, DispatchError> {
-		let asset_pair_account = XYK::get_pair_id(AssetPair::new(insuff_asset_id, asset_out));
+		let asset_pair = AssetPair::new(insuff_asset_id, asset_out);
+		if !XYK::exists(asset_pair) {
+			return Err(DispatchError::Other("XYK pool does not exist for fee payment asset"));
```

Comment:
the text from other is dropped on the runtime, so please change some more descriptive error, ideally `TokenPoolNotFound` from XYK pallet